### PR TITLE
Fix initial dask job state

### PR DESCRIPTION
**Pull request by bstadlbauer**
# TL;DR
When the `dask` plugin creates a `DaskJob` k8s resource, it will initially not have the `.Status.JobStatus` property set. 

So 
https://github.com/flyteorg/flyteplugins/blob/de1d1d345f3bf487701b1f732b2c56f76e81e086/go/tasks/plugins/k8s/dask/dask.go#L316
will return an empty string until the `DaskOperator` sets the status for the first time. While this period is very brief, we've seen it happening a few times.

This PR adds support for the `""` state, effectively handling it the same as being queued/initializing.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

### CODE REVIEW

**Review by hamersaw — DISMISSED** [review]

**Comment by hamersaw** [comment]
@bstadlbauer looks great, lets merge after the lint fix!

**Review by hamersaw — APPROVED** [review]
